libraries/Microsoft.Bot.Builder.AI.Orchestrator/OrchestratorRecognizer.cs

```diff
@@ -172,15 +172,24 @@ public override async Task<RecognizerResult> RecognizeAsync(DialogContext dc, Sc
                     ((List<Result>)recognizerResult.Properties[ResultProperty]).Insert(0, new Result() { Score = 1.0, Label = new Label() { Name = NoneIntent, Type = LabelType.Intent } });
 
                     // add all scores
-                    foreach (var result in results)
+                    bool duplicateNoneRemoved = false;
+                    for (int i = 0; i < results.Count; ++i)
                     {
-                        if (results[0].Label.Name != NoneIntent || result.Label.Name != NoneIntent)
+                        var result = results[i];
+                        string label = result.Label.Name;
+                        if (!duplicateNoneRemoved && label == NoneIntent && result.Score < 1.0)
+                        {
+                            // remove duplicate None intents
+                            results.RemoveAt(i--);
```

Comment:
Is it safe to remove an element from results, while iterating an index to its original results.Count?